src/ray/rpc/raylet/raylet_client.h

```diff
@@ -203,6 +209,9 @@ class RayletClient {
 
   /// Heartbeat timer.
   boost::asio::deadline_timer heartbeat_timer_;
+
+  /// Mutex for thread safe.
```

Comment:
For future reference, anytime we need to add a lock, we should think very carefully about when/whether it is necessary, and the description here should include that information.